r/vignettes/developers/setup.Rmd

```diff
@@ -38,50 +38,49 @@ set -e
 set -x
 ```
 
+The Arrow R package is unique compared to other R packages that you may have
+contributed to because it builds on top of the large and feature-rich Arrow C++
+implementation.
 
-```{bash, save=run & windows, hide=TRUE}
-# For some reason CRAN Mirror goes missing in CI
-echo 'options(repos=structure(c(CRAN="https://cloud.r-project.org")))' > $HOME/.Rprofile
-```
-
-Windows and macOS users who wish to contribute to the R package and
-don't need to alter libarrow (Arrow's C++ library) may be able to obtain a
-recent version of the library without building from source.
-
-### Linux
+## Option 1: Using pre-built libarrow binaries
 
-On Linux, you can download a .zip file containing libarrow from the
+On Linux and MacOS, you can download a .zip file containing libarrow from the
 [nightly repository](https://nightlies.apache.org/arrow/r/libarrow/bin/).
 
 The directory names correspond to the OpenSSL version the binaries built with:
-- "linux-openssl-1.0" (OpenSSL 1.0)
-- "linux-openssl-1.1" (OpenSSL 1.1)
-- "linux-openssl-3.0" (OpenSSL 3.0)
 
-Version numbers in that repository correspond to dates.
+- "linux-openssl-1.0" (OpenSSL 1.0, e.g., Centos7)
+- "linux-openssl-1.1" (OpenSSL 1.1, e.g., Ubuntu 18.04)
+- "linux-openssl-3.0" (OpenSSL >=3.0, e.g., MacOS or Ubuntu >= 20.04)
 
-You'll need to create a `libarrow` directory inside the R package directory and unzip the zip file containing the compiled libarrow binary files into it.
+Version numbers in that repository correspond to dates. Use the version with
+the highest number (i.e., that was modified most frequently).
 
-### macOS
-On macOS, you can install libarrow using [Homebrew](https://brew.sh/):
+You'll need to create a `libarrow` directory inside the R package directory and
+unzip the zip file containing the compiled libarrow binary files into it.
```

Comment:
Doesn't this happen automatically now? I thought that's where we were going when we removed the brew stuff?

Reply:
Ah, we can use `ARROW_DOWNLOADED_BINARIES` on mac and linux and `RWINLIB_LOCAL` on windows to point to the downloaded zip. The script will then unzip into r/libarrow.

With the other changes taking a while to validate on ci I might be able to finish the latestnightly stuff before the release, otherwise shortly after (which would also be fine imo as it only applies to a dev setup)

Reply:
Do you have a ticket already? I would be happy to write one if not. Perhaps we could work on this in parallel to get it over the line quicker. But I don't want to step on something that already exists / is already being worked

Reply:
I think this existing one is applicable: https://github.com/apache/arrow/issues/27839

I have the logic done, I just want to refactor *libs.R a bit as they are overlapping more and more and I don't want to keep copying stuff across ^^ I should be able to get it done today... I will open a PR with my changes regardless of how far I am later today so you can test/review/add to it :)

Reply:
Here is the PR with the latest nightly search + some bonus refactoring ^^)https://github.com/apache/arrow/pull/38236